src/main/java/com/iexec/core/metric/MetricService.java

```diff
@@ -37,6 +53,23 @@ public MetricService(DealWatcherService dealWatcherService,
         this.taskService = taskService;
     }
 
+    //The PostConstruct will be invoked during the bean initialization.
+    @PostConstruct
+    void initializeMetrics() {
+        aliveWorkersCount = Metrics.gauge("iexec.core.workers", new AtomicInteger(workerService.getAliveWorkers().size()));
+        aliveTotalCpuCount = Metrics.gauge("iexec.core.cpu.total", new AtomicInteger(workerService.getAliveTotalCpu()));
+        aliveAvailableCpuCount = Metrics.gauge("iexec.core.cpu.available", new AtomicInteger(workerService.getAliveAvailableCpu()));
+        completedTasksCount = Metrics.gauge("iexec.core.tasks.completed", new AtomicInteger(taskService.findByCurrentStatus(TaskStatus.COMPLETED).size()));
+        dealEventsCount = Metrics.gauge("iexec.core.deals.events", new AtomicLong(dealWatcherService.getDealEventsCount()));
+        dealsCount = Metrics.gauge("iexec.core.deals", new AtomicLong(dealWatcherService.getDealsCount()));
+        replayDealsCount = Metrics.gauge("iexec.core.deals.replay", new AtomicLong(dealWatcherService.getReplayDealsCount()));
+        if (null != dealWatcherService.getLatestBlockNumberWithDeal()) {
+            latestBlockNumberWithDeal = Metrics.gauge("iexec.core.deals.last.block", new AtomicLong(dealWatcherService.getLatestBlockNumberWithDeal().longValue()));
+        } else {
+            latestBlockNumberWithDeal = Metrics.gauge("iexec.core.deals.last.block", new AtomicLong(-1L));
+        }
```

Comment:
We could avoid a double call to `getLatestBlockNumberWithDeal`:
```suggestion
        BigInteger latestBlockNumberWithDealBigInt = dealWatcherService.getLatestBlockNumberWithDeal();
        latestBlockNumberWithDeal = Metrics.gauge("iexec.core.deals.last.block", new AtomicLong(latestBlockNumberWithDealBigInt == null ? -1L : latestBlockNumberWithDealBigInt));
```

The var name may not be the best choice there ;)

Reply:
I can cheat and rename all my gauges with the metric suffix, so I can use the correct variable name

Reply:
or maybe like that

        latestBlockNumberWithDeal = Metrics.gauge(METRIC_DEALS_LAST_BLOCK,
                new AtomicLong(Optional.ofNullable(dealWatcherService.getLatestBlockNumberWithDeal()).orElse(BigInteger.valueOf(-1)).longValue()));

Reply:
Something is looking weird to me here. `latestBlockNumberWithDeal` is initialized at `BigInteger.ZERO`.
I have the impression it should never be `null`. By convention, our APIs seems to return 0 instead of -1 when the blockchain is not reachable.
For those kind of metrics, why don't we just expose them in the correct service and scrap them to create exposed metric when the MetricsController is actually called.

For example, I think the latest block number metric should be managed in DealWatcherService. It should be updated every time a valid deal is received and be fetched by metrics collection service when they perfom their scrape action.
What do you think ?